qa/standalone/test_ceph_daemon.sh

```diff
@@ -1,33 +1,38 @@
 #!/bin/bash -ex
 
-# respawn ourselves with a modified path with both python2 and python3
+FSID='00000000-0000-0000-0000-0000deadbeef'
+IMAGE='ceph/daemon-base:latest-master-devel'
+
+[ -z "$SUDO" ] && SUDO=sudo
+
+if [ -z "$CEPH_DAEMON" ]; then
+    [ -x ../src/ceph-daemon ] && CEPH_DAEMON=../src/ceph-daemon
+    [ -x ./ceph-daemon ] && CEPH_DAEMON=.ceph-daemon
+    which ceph-daemon && CEPH_DAEMON=$(which ceph-daemon)
+fi
+
+# respawn ourselves with a shebang
 PYTHONS="python3 python2"  # which pythons we test
 if [ -z "$PYTHON_KLUDGE" ]; then
    TMPBINDIR=`mktemp -d $TMPDIR`
    trap "rm -rf $TMPBINDIR" TERM HUP INT
-
+   ORIG_CEPH_DAEMON="$CEPH_DAEMON"
+   CEPH_DAEMON="$TMPBINDIR/ceph-daemon"
    for p in $PYTHONS; do
-       ln -s `which $p` $TMPBINDIR/python
        echo "=== re-running with $p ==="
-       PYTHON_KLUDGE=1 PATH=$TMPBINDIR:$PATH $0
+       ln -s `which $p` $TMPBINDIR/python
+       echo "#!$TMPBINDIR/python" > $CEPH_DAEMON
+       cat $ORIG_CEPH_DAEMON >> $CEPH_DAEMON
```

Comment:
Minor nit - 
```suggestion
       tail -n +2 $ORIG_CEPH_DAEMON >> $CEPH_DAEMON
```

The temp ceph-daemon script is created with two shebangs ...
```
$ head -4 /tmp/tmp.t91Qdk20Dy/ceph-daemon
#!/tmp/tmp.t91Qdk20Dy/python
#!/usr/bin/python

DEFAULT_IMAGE='ceph/daemon-base'
DATA_DIR='/var/lib/ceph'
```
... which looks odd, but appears to be functionally correct.